lib/Differentiator/ReverseModeVisitor.cpp

```diff
@@ -79,8 +79,149 @@ namespace clad {
+      auto* VD =
+          ParmVarDecl::Create(m_Context, gradientOverloadFD, noLoc, noLoc,
+                              GVD->getIdentifier(), GVD->getType(),
+                              GVD->getTypeSourceInfo(), GVD->getStorageClass(),
+                              // Clone default arg if present.
+                              (GVD->hasDefaultArg()
+                                   ? Clone(GVD->getDefaultArg())
+                                   : nullptr));
+      if (VD->getIdentifier())
+        m_Sema.PushOnScopeChains(VD, getCurrentScope(),
+                                 /*AddToContext=*/false);
+      overloadParams.push_back(VD);
+      callArgs.push_back((Expr*)BuildDeclRef(VD));
+    }
+
+    for (size_t i = 0; i < remainingArgs; i++) {
+      IdentifierInfo* DVDII = &m_Context.Idents.get("_d_" + std::to_string(i));
+
+      auto DVD = ParmVarDecl::
+          Create(m_Context, gradientOverloadFD, noLoc, noLoc, DVDII,
+                 DerivedOutputParamType,
+                 m_Context.getTrivialTypeSourceInfo(DerivedOutputParamType,
+                                                    noLoc),
+                 StorageClass::SC_None,
+                 // Clone default arg if present.
+                 /*DefArg=*/nullptr);
+      if (DVD->getIdentifier())
+        m_Sema.PushOnScopeChains(DVD, getCurrentScope(),
+                                 /*AddToContext=*/false);
+      overloadParams.push_back(DVD);
+    }
+
+    llvm::ArrayRef<ParmVarDecl*> overloadParamsRef =
+        llvm::makeArrayRef(overloadParams.data(), overloadParams.size());
+
+    llvm::MutableArrayRef<Expr*> callArgsRef =
+        llvm::makeMutableArrayRef(callArgs.data(), callArgs.size());
+
+    gradientOverloadFD->setParams(overloadParamsRef);
+    gradientOverloadFD->setBody(/*B=*/nullptr);
+
+    beginScope(Scope::FnScope | Scope::DeclScope);
+    m_DerivativeFnScope = getCurrentScope();
+    beginBlock();
+
+    Expr* callExpr = nullptr;
+    if (auto* gradientCMD = dyn_cast<CXXMethodDecl>(GradientFD)) {
+      Expr* thisExpr = clad_compat::Sema_BuildCXXThisExpr(m_Sema, gradientCMD);
+      // FIXME: Find a method to do this in a consistent way across all clang
+      //  versions.
+      // BuildMemberReferenceExpr returns a dependent type which is incompatible
+      // with BuildCallToMemberFunction
+#if CLANG_VERSION_MAJOR <= 9
+      // This method is unable to lookup the GradientFD in clang versions 10
+      // and above
+      callExpr = BuildCallExprToMemFn(thisExpr,
+                                      /*isArrow=*/true, GradientFD->getName(),
+                                      callArgsRef);
+#else
+      // NOUR_None is not supported in clang versions 6 and below
+      auto memberExpr =
+          MemberExpr::Create(m_Context, thisExpr, true, noLoc,
+                             NestedNameSpecifierLoc(gradientCMD->getQualifier(),
+                                                    nullptr),
```

Comment:
```suggestion
                                                    /*Data=*/nullptr),
```
Could you move this in a separate variable?